client/jetpack-connect/plans-landing.jsx

```diff
@@ -19,22 +21,48 @@ class PlansLanding extends Component {
 	static propTypes = {
 		basePlansPath: PropTypes.string,
 		interval: PropTypes.string,
+		url: PropTypes.string,
+	};
+
+	static defaultProps = {
+		url: '',
 	};
 
 	componentDidMount() {
 		this.props.recordTracksEvent( 'calypso_jpc_plans_landing_view', {
 			jpc_from: 'jetpack',
 		} );
+
+		this.goToPlansIfAlreadyConnected();
```

Comment:
Let's leave it is `componentDidMount` in case we ever decide to make this section SSR-friendly 😉